src/testing/mock.md

```diff
@@ -1,218 +1,310 @@
+
+## Storage Changes
 
-To test the module, construct a mock runtime. To do so, create a configuration type called `Test` which implements the configuration traits.
+Changes to storage can be checked by direct calls to the storage values. The syntax is the same as it would be in the module's runtime methods
 
 ```rust
-#[derive(Clone, PartialEq, Eq, Debug)]
-pub struct Test;
+#[test]
+fn last_value_updates() {
+	ExtBuilder::build().execute_with(|| {
+	HelloSubstrate::set_value(Origin::signed(1), 10u64);
+	assert_eq!(HelloSubstrate::last_value(), 10u64);
+	HelloSubstrate::set_value(Origin::signed(2), 11u64);
+	assert_eq!(HelloSubstrate::last_value(), 11u64);
+	// .. more assert statements
+	})
+}
 ```
 
-The [derive attribute](https://doc.rust-lang.org/edition-guide/rust-2018/macros/custom-derive.html) ensures that you don't have to manually implement the `Clone, PartialEq, Eq, Debug` traits; the compiler does this for you thereby ensuring that the `Test` type conforms to the behavior of these traits.
-
-Even so, this doesn't work for all traits. Indeed, there are a few traits that require manual implementation to set up the testing environment in `test.rs`. In most case, these *implementations* are limited to specifying the type in your module that corresponds to the type in the imported module. For example, the [Staking module](https://github.com/paritytech/substrate/blob/master/srml/staking/src/mock.rs) implements the `balances` trait in its `mock.rs` file like so:
+For context, the tested module's `decl_storage` block looks like
 
 ```rust
-impl balances::Trait for Test {
-	type Balance = u64;
-	type OnFreeBalanceZero = Staking;
-	type OnNewAccount = ();
-	type Event = ();
-	type TransactionPayment = ();
-	type TransferPayment = ();
-	type DustRemoval = ();
+decl_storage! {
+	trait Store for Module<T: Trait> as HelloSubstrate{
+		pub LastValue get(fn last_value): u64;
+		pub UserValue get(fn user_value): map T::AccountId => u64;
+	}
 }
 ```
 
-> If the configuration logic is not overly complicated, the pattern that follows below can be forgone and replaced with something like [the test scaffolding in `srml/aura`](https://github.com/paritytech/substrate/blob/master/srml/aura/src/mock.rs).
+Updates to `UserValue` are *covered* in the `last_value_updates` test in [`kitchen/module/hello-substrate`](https://github.com/substrate-developer-hub/recipes/tree/master/kitchen/modules/hello-substrate).
+
+## `impl system::Trait`
 
-Next, define an `ExtBuilder` struct that contains the configuration items from your module. In [`srml/staking`](https://github.com/paritytech/substrate/blob/master/srml/staking/src/mock.rs), this looks like
+Substrate's design makes it convenient for the `trait Trait` to inherit `system::Trait` to inherit its types (*[remember](https://substrate.dev/recipes/traits/index.html)*). To inherit this behavior in the mock `Runtime`, it is necessary to `impl` the `system::Trait` (and import some types to do so),
```

Comment:
More than convenient. I think this is necessary because `pub trait Trait: system::Trait`.

Reply:
true, reworded